plugins/housekeeping/msd-ldsm-dialog.c

```diff
@@ -160,7 +160,7 @@ ignore_check_button_toggled_cb (GtkToggleButton *button,
 
         settings_list = g_settings_get_strv (settings, SETTINGS_IGNORE_PATHS);
 
-        for (i = 0; i < G_N_ELEMENTS (settings_list); i++) {
+        for (i = 0; i < g_strv_length (settings_list); i++) {
                 if (settings_list[i] != NULL)
                         ignore_paths = g_slist_prepend (ignore_paths, g_strdup (settings_list[i]));
         }
```

Comment:
A Strv is `NULL`-terminated, so there is a redundant check here (not very important though).  It could be simplified and optimized like this:

```c
for (i = 0; settings_list && settings_list[i]; i++) {
    ignore_paths = g_slist_prepend (ignore_paths, g_strdup (settings_list[i]));
}
```
or by just dropping the `settings_list[i] != NULL` check.

Reply:
Just FYI, as `g_strv_length()` computes the length of the array by traversing it, the code `for (i = 0; i < g_strv_length (settings_list); i++)` has `O(n²)` complexity while `for (i = 0; settings_list && settings_list[i]; i++)` has `O(n)`.

However, it most probably really doesn't matter here and it's arguable that `g_strv_length()` reads better.